Feat/4106 UI implement emissions main view layout

fanwari1 opened the pull request:
* Adds layouts for emissions within HeaderInfo
* Added MultiSelectCombobox component
* Added PillButton component, needed by MultiSelectCombobox
* Added fortawesome icon lib

fanwari1 committed:
Added layout for emissions in HeaderInfo

fanwari1 committed:
Added MultiSelectCombobox component

Implement MultiSelectCombobox component for year and quarter dropdowns

Added PillButton component meant to be used with MultiSelectCombobox

Added fortawesome icon library

fanwari1 committed:
Updated fortawesome lib to latest version

Code cleanup

fanwari1 committed:
Merge branch 'develop' into feat/4106-ui-implement-emissions-main-view-layout

eglove commented on src/components/EmissionsTab/EmissionsTab.js:
```diff
@@ -0,0 +1,126 @@
+    setCurrentTabIndex(getCurrentTabIndex())
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  },[selectedConfig,checkout,currentTabIndex]);
```
effects aren't needed for computed values. This is going to set on every render regardless. Using an effect here makes it run twice. Memo is not necessary either because the calculation isn't expensive enough for it to be faster. (Memo runs a deep comparison that would slow this down.)

All that's needed is const currentTabIndex = getCurrentTabIndex()

eglove commented on src/components/EmissionsTab/EmissionsTab.js:
```diff
@@ -0,0 +1,126 @@
+  useEffect(() => {
+    setLocation(locationSelect, title, workspaceSection);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [locationSelect]);
```
We'll not bother with the fact that there's a state setter sitting somewhere in Redux. I don't think there's a comfortable way to fix this.

eglove commented on src/components/MultiSelectCombobox/MultiSelectCombobox.js:
```diff
@@ -0,0 +1,256 @@
+    populateSelectedItems();
+    return () => setStillMounted(false);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```
populateSelectedItems is going to be recreated on every render. Even if it doesn't change, a new one exists on every instance of the component.

There's no reason for this eslint ignore, it should contain [populateSelectedItems] and the effect can be moved below all the declared functions.

fanwari1 replied:
Updated

eglove commented on src/components/MultiSelectCombobox/MultiSelectCombobox.js:
```diff
@@ -0,0 +1,256 @@
+        ) {
+          setShowListBox(false);
+        }
+      })}
```
Double check that this isn't a memory leak, if this event listener is being added on every render but not removed it was cause a major performance drag.

Event listeners should be added in effects and removed in their returns.

fanwari1 replied:
It does indeed and code has been updated to clean up the event.

fanwari1 committed:
Added cleanup of click event handler.

Removed eslint supressor and updated dep array for pupulateSelectedItems useeffect

fanwari1 committed:
Removed useless useeffect for setting tab index

fanwari1 committed:
Fixed code smells
